sniffles/feature.py

```diff
@@ -359,27 +418,73 @@ def parseLine(self, line=None):
                 return None
             if mypairs['type'].lower() == 'feature':
                 myfeature = Feature(name, lower_bound, upper_bound,
-                                    complexity_prob, myambiguity_list)
+                                    complexity_prob, ambiguity_list)
             elif mypairs['type'].lower() == 'content':
                 myfeature = ContentFeature(name, regex, complexity_prob, len)
             elif mypairs['type'].lower() == 'ip':
                 myfeature = IPFeature(name, version, complexity_prob)
             elif mypairs['type'].lower() == 'protocol':
                 myfeature = ProtocolFeature(name, proto_list, complexity_prob,
-                                            myambiguity_list)
+                                            ambiguity_list)
             else:
                 print("Unrecognized feature type.", line)
                 return None
             self.features.append(myfeature)
 
+    def tokenizeAmbiguityList(self, list):
+        notTrim = list[1:-1]
+        parsedlist = ""
+        # remove all space
+        # currently, sniffles support no space
+        for i in range(0, len(notTrim)):
+            if notTrim[i] != " ":
+                parsedlist += notTrim[i]
+        values = []
+        currentIndex = 0
+        beginIndex = 0
+        lastIndex = len(parsedlist) - 1
+        while currentIndex <= lastIndex:
+            if parsedlist[currentIndex] == ",":
+                tmpStr = parsedlist[beginIndex: currentIndex]
+                values.append(tmpStr)
+                currentIndex += 1
+                beginIndex = currentIndex
+            elif parsedlist[currentIndex] == "[":
+                beginIndex = currentIndex
+                while parsedlist[currentIndex] != "]":
+                    currentIndex += 1
+                currentIndex += 1
+                tmpStr = parsedlist[beginIndex: currentIndex]
+                values.append(tmpStr)
+                currentIndex += 1
+                beginIndex = currentIndex
+            elif parsedlist[currentIndex] == "{":
+                beginIndex = currentIndex
+                while parsedlist[currentIndex] != "}":
+                    currentIndex += 1
+                currentIndex += 1
+                tmpStr = parsedlist[beginIndex: currentIndex]
+                values.append(tmpStr)
+                currentIndex += 1
+                beginIndex = currentIndex
+            else:
+                currentIndex += 1
+                if currentIndex > lastIndex and \
+                   currentIndex > beginIndex:
```

Comment:
You should check this statement.  I don't think we need the second condition.  If currentIndex is greater than lastIndex, then it should be greater than or equal to the beginIndex, right?